pkg/operator/controllers/dnsmasq/cluster_controller_test.go

```diff
@@ -0,0 +1,129 @@
+package dnsmasq
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"strconv"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	mcv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	"github.com/sirupsen/logrus"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	ctrlfake "sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	mock_dynamichelper "github.com/Azure/ARO-RP/pkg/util/mocks/dynamichelper"
+)
+
+func TestClusterReconciler(t *testing.T) {
+	fakeDh := func(controller *gomock.Controller) *mock_dynamichelper.MockInterface {
+		return mock_dynamichelper.NewMockInterface(controller)
+	}
+	cluster := func(enabled bool) *arov1alpha1.Cluster {
+		return &arov1alpha1.Cluster{
+			ObjectMeta: metav1.ObjectMeta{Name: "cluster"},
+			Status:     arov1alpha1.ClusterStatus{},
+			Spec: arov1alpha1.ClusterSpec{
+				OperatorFlags: arov1alpha1.OperatorFlags{
+					controllerEnabled: strconv.FormatBool(enabled),
+				},
+			},
+		}
+	}
+
+	t.Run("when no cluster resource is present, returns error", func(t *testing.T) {
```

Comment:
I wish we could use tabular tests here to remain consistent with the rest of our unit tests.

Reply:
I have brought back the tabular tests for these controllers.